.github/workflows/_test_rosetta.yaml

```diff
@@ -14,16 +14,52 @@ on:
 
 jobs:
   rosetta-tests:
-    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        MARKERS: ["", "-m integration"]
+      fail-fast: false
+    runs-on: [self-hosted, compute, V100]
     steps:
       - name: Print environment variables
         run: |
           env
 
+      - name: Print GPU information
+        run: nvidia-smi  
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Pull Rosetta image
+        shell: bash -x -e {0}
+        run: |
+          docker pull ${{ inputs.ROSETTA_IMAGE }}
+
+      - name: Run Rosetta tests w/ docker
+        run: |
+          set -x
+          docker run --gpus '"device=0,1"' ${{ inputs.ROSETTA_IMAGE }} sh -c "pip install '/opt/rosetta[test]' && pytest /opt/rosetta ${{ matrix.MARKERS }}"
+
   rosetta-lint-check:
```

Comment:
It just occurred to me that we should lint Rosetta before even the build, right?

Reply:
We can create a separate reusable workflow just for linting either rosetta or even the entire repo, and call it at the beginning of the entire CI pipeline.

Reply:
How about we leave linting out of the scope of this PR, and do the whole-repo linting via another PR?

Reply:
> It just occurred to me that we should lint Rosetta before even the build, right?

Good point, it could be good to do it before since we'd avoid wasting compute on improperly formatted code. I originally created it as a parallel step since I would imagine we'd be interested in the functionality even if the code wasn't to our style guidelines.

I can remove it for now since it's not a hard change to re-introduce (and there's nothing substantive to lint yet anyways 😄)

Reply:
Sounds good. We can enforce style guide later.